e2e/nomostest/config_sync.go

```diff
@@ -947,17 +931,66 @@ func RepoSyncObjectV1Beta1(nn types.NamespacedName, repoURL string, sourceFormat
+// RootSyncObjectGit creates a new RootSync object with a Git source, using the
+// default test config plus the specified inputs.
+// Creates, updates, or replaces the SyncSource expectation for this RootSync.
+func (nt *NT) RootSyncObjectGit(name string, repo *gitproviders.Repository, syncPath string, sourceFormat configsync.SourceFormat) *v1beta1.RootSync {
+	id := core.RootSyncID(name)
+	SetExpectedGitSource(nt, id, repo, syncPath, sourceFormat)
+	repoURL := nt.GitProvider.SyncURL(repo.RemoteRepoName)
+	rs := RootSyncObjectV1Beta1Git(name, repoURL, sourceFormat)
+	if nt.DefaultReconcileTimeout != nil {
+		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*nt.DefaultReconcileTimeout)
+	} else if rs.Spec.Override != nil {
+		rs.Spec.Override.ReconcileTimeout = nil
 	}
+	// Enable automatic deletion of managed objects by default.
+	// This helps ensure that test artifacts are cleaned up.
+	EnableDeletionPropagation(rs)
+	return rs
+}
 
-	rs := RootSyncObjectV1Beta1FromRootRepo(nt, name)
+// RepoSyncObjectGit creates a new RepoSync object with a Git source, using the
+// default test config plus the specified inputs.
+// Creates, updates, or replaces the SyncSource expectation for this RepoSync.
+func (nt *NT) RepoSyncObjectGit(nn types.NamespacedName, repo *gitproviders.Repository, syncPath string, sourceFormat configsync.SourceFormat) *v1beta1.RepoSync {
+	id := core.RepoSyncID(nn.Name, nn.Namespace)
+	SetExpectedGitSource(nt, id, repo, syncPath, sourceFormat)
+	repoURL := nt.GitProvider.SyncURL(repo.RemoteRepoName)
+	// RepoSync is always Unstructured. So ignore repo.Format.
+	rs := RepoSyncObjectV1Beta1Git(nn, repoURL, sourceFormat)
+	if nt.DefaultReconcileTimeout != nil {
+		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*nt.DefaultReconcileTimeout)
+	} else if rs.Spec.Override != nil {
+		rs.Spec.Override.ReconcileTimeout = nil
+	}
+	// Add dependencies to ensure managed objects can be deleted.
+	if err := SetRepoSyncDependencies(nt, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+	// Enable automatic deletion of managed objects by default.
+	// This helps ensure that test artifacts are cleaned up.
+	EnableDeletionPropagation(rs)
+	return rs
+}
+
+// RootSyncObjectOCI returns a RootSync object that syncs the provided OCIImage.
+// ImageID is used for pulling. ImageDigest is used for validating the result.
+// Creates, updates, or replaces the SyncSource expectation for this RootSync.
+func (nt *NT) RootSyncObjectOCI(name string, imageID registryproviders.OCIImageID, syncPath, expectedImageDigest string) *v1beta1.RootSync {
+	id := core.RootSyncID(name)
+	// Register the OCI image for syncing
+	expectedSyncPath := syncPath
+	if syncPath == "" {
+		expectedSyncPath = controllers.DefaultSyncDir
+	}
+	SetExpectedOCISource(nt, id, imageID, expectedImageDigest, expectedSyncPath)
+	rs := k8sobjects.RootSyncObjectV1Beta1(name)
+	rs.Spec.SourceFormat = configsync.SourceFormatUnstructured // TODO: is SourceFormat necessary with OCI?
```

Comment:
Technically, OCI source configs can use the hierarchical format. We don't have test coverage for that scenario. Given that unstructured is the recommended format, so we can skip testing OCI with hierarchical format.

Reply:
Replaced TODO with a comment in https://github.com/GoogleContainerTools/kpt-config-sync/pull/1410